cli/internal/util.go

```diff
@@ -161,3 +161,40 @@ func ParseKeyPair(p []string) map[string]string {
 	}
 	return params
 }
+
+/*
+ParseStep parses the step id form a string which may eigher be the step PID (step number) or a step name.
+These rules apply:
+
+  - Step PID take precedence over step name when searching for a match.
+  - First match is used, when there are multiple steps with the same name.
+
+Strictly speaking, this is not parsing, but a lookup.
+*/
+func ParseStep(client woodpecker.Client, repoID, number int64, stepArg string) (stepID int64, err error) {
+	pipeline, err := client.Pipeline(repoID, number)
+	if err != nil {
+		return 0, err
+	}
+
+	stepPID, err := strconv.ParseInt(stepArg, 10, 64)
+	if err == nil {
+		for _, wf := range pipeline.Workflows {
+			for _, step := range wf.Children {
+				if int64(step.PID) == stepPID {
+					return step.ID, nil
+				}
+			}
+		}
+	}
+
+	for _, wf := range pipeline.Workflows {
+		for _, step := range wf.Children {
+			if step.Name == stepArg {
+				return step.ID, nil
+			}
+		}
+	}
```

Comment:
I intentionally did not do it this way (had it like that before), because if your second  step is named "5" the outcome might not what you would expect if you provide 5 as an argument.

At least I dicded that step PID has precedence, so you canaddress any step.

Edge case, but ...